mne/label.py

```diff
@@ -416,27 +414,34 @@ def fill(self, src, name=None):
         label : Label
             The label covering the same vertices in source space but also
             including intermediate surface vertices.
+
+        See Also
+        --------
+        Label.restrict
+        Label.smooth
         """
         # find source space patch info
         if len(self.vertices) == 0:
             return self.copy()
-        if self.hemi == 'lh':
-            hemi_src = src[0]
-        elif self.hemi == 'rh':
-            hemi_src = src[1]
+        hemi_src = _get_label_src(self, src)
 
         if not np.all(np.in1d(self.vertices, hemi_src['vertno'])):
             msg = "Source space does not contain all of the label's vertices"
             raise ValueError(msg)
 
-        nearest = hemi_src['nearest']
-        if nearest is None:
-            warn("Computing patch info for source space, this can take "
-                 "a while. In order to avoid this in the future, run "
+        if hemi_src['nearest'] is None:
+            warn("Source space is being modified in place because patch "
+                 "information is needed. To avoid this in the future, run "
                  "mne.add_source_space_distances() on the source space "
-                 "and save it.")
-            add_source_space_distances(src)
-            nearest = hemi_src['nearest']
+                 "and save it to disk.")
+            if check_version('scipy', '1.3'):
+                dist_limit = 0
+            else:
+                warn('SciPy < 1.3 detected, adding source space patch '
+                     'information will be slower. Consider upgrading SciPy.')
+                dist_limit = np.inf
```

Comment:
can you explain the logic here? 0 if good scipy and inf otherwise with just a warning that it will be slow? it seems you also change the behavior depending on scipy version.

Reply:
Here we just need `s['nearest']`. On SciPy 1.3+ you can do this very quickly (< 1 s) with some new `kwargs`. On old SciPy you have to do the all-sources-to-all-sources calculation (slow, > 1 min).

There is a change of behavior in the sense that doing it the slow way will also populate `src['dists']` in place, so someone who then saves out this `src` after running this function will or won't have `src['dists']` depending on their SciPy version. This seems rare enough not to worry about, but if it seems too dangerous, I can 1) just get rid of the special casing here and restore the always-slow version, or 2) raise an error if the version isn't new enough. I actually prefer (2) in this case because the auto-calculation when it's super slow is actually a bit frustrating, better to have the user calculate and save on their own.

Reply:
ok fair enough. ok for me as it is